core/src/main/java/org/testcontainers/images/PullPolicy.java

```diff
@@ -1,22 +1,62 @@
 package org.testcontainers.images;
 
+import com.google.common.annotations.VisibleForTesting;
 import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+import org.testcontainers.utility.TestcontainersConfiguration;
 
 import java.time.Duration;
 
 /**
  * Convenience class with logic for building common {@link ImagePullPolicy} instances.
  *
  */
+@Slf4j
 @UtilityClass
 public class PullPolicy {
 
+    @VisibleForTesting
+    static ImagePullPolicy instance;
+
+    @VisibleForTesting
+    static ImagePullPolicy defaultImplementation = new DefaultPullPolicy();
+
     /**
      * Convenience method for returning the {@link DefaultPullPolicy} default image pull policy
      * @return {@link ImagePullPolicy}
      */
     public static ImagePullPolicy defaultPolicy() {
-        return new DefaultPullPolicy();
+        if (instance == null) {
```

Comment:
1. Are we okay with double instantiation when it is used concurrently? Or should we mark it as `synchronized`?
2. minor nit: consider doing an early return on `instance != null` to reduce the indentation 👍

Reply:
thanks! I've applied changes and add docs too.